### PR TITLE
add touch events on touch monitor #261

### PR DESCRIPTION
**Pull request by urbanekpj**
see #261
on my touch device i can't 'drag' (eg. select text). i added onTouch events to onMouse events and it is working now. but im not sure if it is relay not working or it should work but my setup is broken.

### CODE REVIEW

**Comment by m1k1o** [comment]
It's not broken for you, instead there is no touch support yet. Thanks for adding it!

File: client/src/components/video.vue [hunk 1]
```diff
@@ -678,6 +681,46 @@
...
+          break
+        case 'touchend':
+          type = 'mouseup'
+          break
+        default:
+          return
+      }
+
+      const simulatedEvent = document.createEvent('MouseEvent')
+      simulatedEvent.initMouseEvent(
```
Comment (m1k1o) [inline client/src/components/video.vue:703]: This event seems to be [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/initMouseEvent).

**Comment by m1k1o** [comment]
Touch events should be added as `{ passive: false }` listeners, according to this: https://stackoverflow.com/a/37721906 But vue.js [does not seem to support that](https://github.com/vuejs/vue/issues/10778) yet.

File: client/src/components/video.vue (hunk 1 above)
Comment (m1k1o) [reply client/src/components/video.vue:703]: This seems to be replacement:

```bash
var mouseEvent = new MouseEvent('mousedown', {
  clientX: touch.clientX,
  clientY: touch.clientY
});
```

File: client/src/components/video.vue [hunk 2]
```diff
@@ -678,6 +681,46 @@
...
+        false,
+        false,
+        false,
+        false,
+        0 /*left*/,
+        null,
+      )
+
+      first.target.dispatchEvent(simulatedEvent)
+      event.preventDefault()
```
Comment (m1k1o) [inline client/src/components/video.vue:722]: We already have `.stop.prevent` in the vuejs event, so this line can be removed.

**Comment by m1k1o** [comment]
This is [novnc's implementation](https://github.com/novnc/noVNC/blob/8decca7353a0a730226842aa33c9018ac5e6e44c/core/rfb.js#L1228), quite extended.

**Comment by urbanekpj** [comment]
hi i fixed two minor issues. don'y know what should be done with this prevent( add listeners manually or simply ignore it for now?) 

and as you raised, touch events are more complex. my changes simply map touch to mouse. which is not touchscreen support. (my current changes are sufficient for my use case ) 

full support require more integration on server side also. 
i'm not sure if im capable to adding this full support on browser site not mentioning changes in back end code. 
i'm on edge if this PR is valid. on one hand it is not worse than current support ( it adds select and drag, draw support, what i needed) but is not consistent with general touch user experience.   
what your take on this?

**Comment by m1k1o** [comment]
Also novnc does not seem to use `{ prevent: false }` in their event listeners, so we can leave it out for now and will see.

Given example from novnc is only client side code, they try to convert touch events to mouse events. So only client needs to be altered. Adding full touch passthrough with touch app on the server side would be another approach, yes.

It is good to have at least some touch support, and gestures support would be considered as another feature, so yes your changes are welcome. You can push changes if you have some.

**Comment by urbanekpj** [comment]
i pushed changes with replacing deprecated function. (I forgot to push it before)

**Pull request merged**